.github/workflows/CI.yml

```diff
@@ -135,33 +143,34 @@ jobs:
         with:
           repo-token: ${{ secrets.GITHUB_TOKEN }}
 
-      - name: Cache
-        uses: actions/cache@v4
-        with:
-          path: |
-            ~/.cargo/bin/
-            ~/.cargo/registry/index/
-            ~/.cargo/registry/cache/
-            ~/.cargo/git/db/
-            target/
-          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+      - name: Build Test
+        run: cargo test -p ${{ matrix.package }} --no-run
 
       - name: Run unit and integration tests
         env:
-          DATABASE_URL: "postgres://postgres:postgres@localhost:5432/postgres"
+          DATABASE_URL: "postgres://postgres:postgres@postgres:5432/postgres"
         run: cargo test -p ${{ matrix.package }}  -- --include-ignored
-  
-  tests:
-    needs: build
+
+  build-release:
```

Comment:
what's the purpose of this new separate "build-release" stage compared to the one that actually packages the built binary into a deb package and pushes it up to the packagecloud repo?

Reply:
so to push release you need tests, clippy and all to run so to avoid waiting too long we start building release on start so that by the time test are done we should only have a few min left on build and it can be pulled from cache to upload mostly